src/Esfa.Vacancy.Application/Commands/CreateApprenticeship/CreateApprenticeshipParametersMapper.cs

```diff
@@ -55,14 +56,40 @@ public CreateApprenticeshipParametersMapper(IWageTypeMapper wageTypeMapper, IDur
                 EmployersApplicationInstructions = request.ExternalApplicationInstructions,
                 ContactName = request.ContactName,
                 ContactEmail = request.ContactEmail,
-                ContactNumber = request.ContactNumber
+                ContactNumber = request.ContactNumber,
+                TrainingType = (int)request.TrainingType,
+                ApprenticeshipType = GetApprenticeshipType(request.TrainingType, request.EducationLevel)
             };
 
             MapLocationFields(request, employerInformation, parameters);
 
+            MapTrainingCode(request, parameters);
             return parameters;
         }
 
+        private static ApprenticeshipType GetApprenticeshipType(TrainingType trainingType, int educationLevel)
+        {
+            switch (educationLevel)
+            {
+                case 2:
+                    return ApprenticeshipType.Intermediate;
+                case 3:
+                    return ApprenticeshipType.Advance;
+                case 4:
+                    return ApprenticeshipType.Higher;
+                case 5:
+                    return
+                        trainingType == TrainingType.Framework ? ApprenticeshipType.Degree : ApprenticeshipType.Foundation;
+                case 6:
+                    return ApprenticeshipType.Degree;
+                case 7:
+                    return
+                        trainingType == TrainingType.Framework ? ApprenticeshipType.Degree : ApprenticeshipType.Masters;
+                default:
+                    return (ApprenticeshipType)0;
+            }
+        }
+
```

Comment:
As you have this mapping code here which is the right place - is it worth moving WageTypeMapper and DurationTypeMapper into this class as static methods?  Then we are consistent

Reply:
Since this is mapping code I don't see why we should put it in a separate file, however Scott may disagree. May be we can have a quick discussion and agree on approach. I will leave this as is till then.